src/provisioner/cluster.py

```diff
@@ -124,17 +132,21 @@ def _get_storage_kv_list(self, storage_spec: dict, node_id: str):
 
     def save(self, cortx_conf):
         """ Saves cluster information onto the conf store """
-
         kvs = []
         try:
             for node in self._node_list:
-                node_id = node.pop('id')
+                logical_id = node.pop('id')
+                node_id = CortxCluster.get_machine_id(logical_id)
                 key_prefix = f'node>{node_id}'
+                if socket.gethostname() == node['hostname'] :
```

Comment:
Remove this condition. Logic should be common for all pods unconditionally.

Reply:
As discussed in the call, we have to keep the if condition as of now to recognize the current pod with hostname.